aws/lambda/pytorch-auto-revert/pytorch_auto_revert/signal.py

```diff
@@ -349,7 +377,7 @@ def partition_by_autorevert_pattern(self) -> Optional[PartitionedCommits]:
         return PartitionedCommits(failed=failed, unknown=unknown, successful=successful)
 
     def process_valid_autorevert_pattern(
-        self,
+        self, *, bisection_limit: Optional[int] = None
```

Comment:
I am not a big fan of this, you've done this in multiple places and this forces using `s.process_valid_autorevert_pattern(bisection_limit=bisection_limit)` what is not very pythonic.

why use the `*` here?

Reply:
yes, it forces to always specify the name of the parameter, which is a good thing for safety, especially for generic types, like int